trie-db/src/triedb.rs

```diff
@@ -298,16 +297,15 @@ impl<'a, L: TrieLayout> Iterator for TrieDBIterator<'a, L> {
 		while let Some(item) = self.inner.next() {
 			match item {
 				Ok((mut prefix, _, node)) => {
-					let maybe_value = match node.as_ref() {
-						&OwnedNode::Leaf(ref partial, ref value) => {
-						prefix.append(partial);
-							Some(value.as_ref())
+					let maybe_value = match node.node() {
+						Node::Leaf(partial, value) => {
+							prefix.append_partial(partial.right());
```

Comment:
switch to append_partial is an additional cost here, the method is 'slow', I don't know how much it can be responsible for perf not being as expected.

Reply:
I don't think this is any slower than the `append` it was doing before.

Reply:
If I trust the comment it is shifting nibbles when vec is always aligned.

Reply:
Yes, this is true. But this just changes from `NibbleVec::append` to `NibbleVec::append_partial` and `NibbleVec::append` has exactly the same performance issue. It's just that the parameter type is different.